django_setup_configuration/config_settings.py

```diff
@@ -0,0 +1,227 @@
+
+    def __init__(self, *args, update_field_descriptions: bool = False, **kwargs):
+        self.config_fields: list[ConfigField] = []
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        if not getattr(self, "models", None):
+            return
+
+        # add support for custom fields like PrivateMediaField
+        if update_field_descriptions:
+            self.update_field_descriptions()
+
+        for model in self.models:
+            self.create_config_fields(model=model)
+
+    @staticmethod
+    def get_default_value(field: models.Field) -> str:
+        default = field.default
+
+        if default is NOT_PROVIDED:
+            return "No default"
+
+        # needed to make `generate_config_docs` idempotent
+        # because UUID's are randomly generated
+        if isinstance(field, models.UUIDField):
+            return "random UUID string"
+
+        # if default is a function, call the function to retrieve the value;
+        # we don't immediately return because we need to check the type first
+        # and cast to another type if necessary (e.g. list is unhashable)
+        if callable(default):
+            default = default()
+
+        if isinstance(default, Mapping):
+            return str(default)
+
+        # check for field type as well to avoid splitting values from CharField
+        if isinstance(field, (JSONField, ArrayField)) and isinstance(default, Sequence):
+            try:
+                return ", ".join(str(item) for item in default)
+            except TypeError:
+                return str(default)
+
+        return default
+
+    @staticmethod
+    def update_field_descriptions() -> None:
+        """
+        Add custom fields + descriptions defined in settings to
+        `basic_field_descriptions`
+        """
+        custom_fields = getattr(settings, "DJANGO_SETUP_CONFIG_CUSTOM_FIELDS", None)
+        if not custom_fields:
+            return
+
+        for mapping in custom_fields:
+            try:
+                field = import_string(mapping["field"])
+            except ImportError as exc:
+                raise ImproperlyConfigured(
+                    "\n\nSomething went wrong when importing {field}.\n"
+                    "Check your settings for django-setup-configuration".format(
+                        field=mapping["field"]
+                    )
+                ) from exc
+            else:
+                description = mapping["description"]
+                basic_field_descriptions[field] = description
+
+    @staticmethod
+    def get_field_description(field: models.Field) -> str:
+        # fields with choices
+        if choices := field.choices:
+            example_values = [choice[0] for choice in choices]
+            return ", ".join(example_values)
+
+        # other fields
+        field_type = type(field)
+        if field_type in basic_field_descriptions.keys():
+            return basic_field_descriptions.get(field_type)
```

Comment:
To fix the type hint of the functions, and `ConfigField` expects a string from this as well
```suggestion
            return basic_field_descriptions.get(field_type, "")
```